Automattic/jetpack
# Emails: Remove jetpack_show_wpcom_inbox_menu filter

#### Changes proposed in this Pull Request:

This Pull Request is removing the filter that was added as part of this [Pull Request](https://github.com/Automattic/jetpack/pull/21150).

#### Does this pull request change what data or activity we track or use?
No, this pull request does not change what data ro activity we track

#### Testing instructions:

For WPCOM simple site:

- Merge the changed files from D68453-code on your sandbox
- Without making any changes on your sandbox, verify that the under Upgrades menu, **_there is_** a new Inbox menu

For WPCOM atomic site:

- Ensure that you have the [Jetpack Beta](https://jetpack.com/download-jetpack-beta/) plugin uploaded and installed on your Atomic test site, as well as a mechanism for running code on the site. (A plugin like [Code Snippets](https://wordpress.org/plugins/code-snippets/) can make this relatively easy.)
- Use Jetpack Beta to switch to this branch: remove/inbox-filter-added-while-developing-inbox-menu
- Refresh the admin page you're on, and verify that below Upgrade menu **_IS STILL_** the Inbox menu regarding any value of the value return by the filter `apply_filters( 'jetpack_show_wpcom_inbox_menu', X )` (where X is true or false)

**Screenshots**
![image](https://user-images.githubusercontent.com/5689927/134323002-054ee80a-ead2-42c3-b734-8541b0635768.png)

Related to {1200182182542585-as-1201164484498589}
